easyDiffractionApp/Logic/Fitter.py

```diff
@@ -184,6 +189,81 @@ def currentMinimizerMethodIndex(self, new_index: int):
         self._current_minimizer_method_index = new_index
         self._current_minimizer_method_name = self.minimizerMethodNames()[new_index]  # noqa: E501
 
+    # Constraints
+
+    @Slot(int, str, str, str, int)
+    def addConstraint(self, dependent_par_idx, relational_operator,
+                      value, arithmetic_operator, independent_par_idx):
+        if dependent_par_idx == -1 or value == "":
+            print("Failed to add constraint: Unsupported type")
+            return
+        # if independent_par_idx == -1:
+        #    print(f"Add constraint: {self.fitablesList()[dependent_par_idx]['label']}{relational_operator}{value}")
+        # else:
+        #    print(f"Add constraint: {self.fitablesList()[dependent_par_idx]['label']}{relational_operator}{value}{arithmetic_operator}{self.fitablesList()[independent_par_idx]['label']}")
+        pars = [par for par in self.fitter.fit_object.get_parameters() if par.enabled]
+        if arithmetic_operator != "" and independent_par_idx > -1:
```

Comment:
minor tip: having separate methods for type1 and type2 `addConstraint` functionality would probably make more sense than the convoluted and I think, error prone, set of conditionals.